app/assets/javascripts/discourse/app/components/header.gjs

```diff
@@ -112,6 +112,23 @@ export default class GlimmerHeader extends Component {
     };
   });
 
+  // needed for the styleguide. for normal use, receiving this parameter is not necessary
+  get currentUser() {
+    return this.args.currentUser === undefined
+      ? this.currentUserService
+      : this.args.currentUser;
+  }
+
+  // needed for the styleguide. for normal use, receiving this parameter is not necessary
+  get topicInfo() {
+    return this.args.topicInfo ?? this.header.topicInfo;
+  }
+
+  // needed for the styleguide. for normal use, receiving this parameter is not necessary
+  get topicInfoVisible() {
+    return this.args.topicInfoVisible ?? this.header.topicInfoVisible;
+  }
+
```

Comment:
Would it make sense to always have these as arguments, and pass them through from `GlimmerSiteHeader`? That way, the logic would be the same for 'real' vs. 'styleguide'

(just a random idea - feel free to say "no, that wouldn't work")